### PR TITLE
Refactoring + fakejuggler binary

### PR DESCRIPTION
**Pull request by abulimov**
* Add .gitignore
* Move job definition into `lib`
* Refactor usage of strings as statuses, now they are consts
* Add `fakejuggler` binary to help with testing of UI/Clients

### CODE REVIEW

File: daemon.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package main
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/leoleovich/3djuggler/lib"
```
Comment (yrk-lab) [inline daemon.go]: Not sure if a separate package is necessary... Can we at leaste have a meaningful name which would play nicely with types/constants in the package? "lib.JobStatus" is sort of too generic.

**Review by yrk-lab — CHANGES_REQUESTED** [review]

**Comment by abulimov** [comment]
@yarikk I've renamed `lib` to `juggler` (best meaningful name I could come up with)

File: main.go [hunk 2]
```diff
@@ -78,7 +67,7 @@ func main() {
 	daemon := &Daemon{
 		timer:  time.NewTimer(0),
 		config: &Config{},
-		job:    &Job{Status: "Waiting for job"},
+		job:    &juggler.Job{Status: "Waiting for job"},
```
Comment (leoleovich) [inline main.go]: Maybe use your new type here?

File: main.go [hunk 3]
```diff
@@ -160,46 +149,45 @@ func main() {
...
-						daemon.UpdateStatus("Sending to printer")
+						daemon.UpdateStatus(juggler.StatusSending)
 					} else {
 						log.Info("Waiting ", gizmostatusfileStat.ModTime().Add(waitingForButtonInterval).Unix()-time.Now().Unix(), " more seconds for somebody to press the button")
 					}
 				} else if daemon.job.Scheduled.After(time.Now()) {
 					log.Info("Waiting ", daemon.job.Scheduled.Unix()-time.Now().Unix(), " more seconds for somebody to press the button")
 				} else {
 					log.Warning("Nobody pressed the button on time")
-					daemon.UpdateStatus("Button timeout")
```
Comment (leoleovich) [inline main.go:195]: This is still needed

**Review by leoleovich — CHANGES_REQUESTED** [review]

**Comment by abulimov** [comment]
Addressed comments

**Review by leoleovich — APPROVED** [review]
T.hanks!

**Pull request merged**